### ISSUE TITLE
Add option to backup scheduling to disable update 'on battery'

### ISSUE THREAD

**Issue by nickmdowson** [jensstein/oandbackup#131]
Hi,

Being able to schedule backups is great, but sometimes I haven't plugged my phone in, and it backups automatically at a point where I'm hoping the battery will last as long as possible.

Would you be able to add an option that would prevent it from running a scheduled backup on battery, i.e. it would only run if the device was charging? This could fit as a tick box within the 'scheduling' UI. I imagine that if at the time of the scheduled backup the phone was not plugged in, it would either wait till the phone was next charged and then run, or would just skip the backup (perhaps presenting a notification to the user), until the next time it was scheduled.

Thanks!
Nick

**Comment by jensstein** [comment]
yes, of course, this is a good idea. schedules should just have a "don't run on battery" button.
it will probably be best to skip the backup entirely instead of waiting for the device to be plugged in. or perhaps a kind of snooze-or-dismiss system.

**Comment by DJCrashdummy** [comment]
IMHO completely skipping backups is not a good idea!
a user (i'm not talking tech-nerds but casual users which perhaps don't setup the device/oandbackup themselves) sets the schedule because he wants a recent backup (e.g. daily at night because he is normally charging his phone at that time). he might check the "don't run on battery" box because he may be somehow concerned about longer days (e.g. partying at the weekend) where the longer battery life is desperately needed.
so far so good... at the latest next night the scheduled backup(s) will hopefully run again.
BUT if his daily routine changes and he only charges his phone at work in the midmorning he will end with no backup at all! :slightly_frowning_face: 

probably waiting for the next time the phone is plugged in is much better - for sure just execute one instance of the same job, if it was "delayed" more than once. maybe (if possible) starting the job(s) not until some (configurable?) minutes after plugging in may be an interesting feature.

because i'm just thinking about it:
would running more than one job at the same time eventually cause problems?

also a notification for a pending delayed backup may be reasonable or at least for if skipped ones if this gets implemented!!!
